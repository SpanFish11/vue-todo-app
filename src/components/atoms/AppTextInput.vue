<script setup>
const props = defineProps({
  placeholder: { type: String, required: false, default: 'Add your task' },
})
</script>

<template>
  <input type="text" class="text-input" :placeholder="props.placeholder" />
</template>

<style scoped lang="scss">
.text-input {
  width: 100%;
  padding: 10px;
  margin-bottom: 20px;
  border-radius: 5px;
  border: 1px solid #ddd;
}
</style>
