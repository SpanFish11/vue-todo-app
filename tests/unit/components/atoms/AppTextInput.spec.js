import { describe, expect, it, beforeEach } from '@jest/globals'
import { shallowMount } from '@vue/test-utils'
import AppTextInput from '@/components/atoms/AppTextInput.vue'

describe('Given the AppTextInput component', () => {
  const propsWithOptional = {
    placeholder: 'placeholder',
    maxlength: 5,
  }

  const propsWithoutOptional = {
    placeholder: 'placeholder',
  }

  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(AppTextInput, {})
  })

  it.each([[propsWithOptional], [propsWithoutOptional]])('should render', (props) => {
    wrapper.setProps(props)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should sets the value', async () => {
    const input = wrapper.find('input')
    await input.setValue('test')

    expect(input.element.value).toBe('test')
  })

  describe('should emit when', () => {
    let input

    beforeEach(() => {
      input = wrapper.find('input')
    })

    it('pressed enter button', async () => {
      await input.trigger('keyup.enter')

      const onPressEnterEvent = wrapper.emitted('onPressEnter')

      expect(wrapper.emitted()).toHaveProperty('onPressEnter')
      expect(onPressEnterEvent).toHaveLength(1)
    })

    it('pressed esc button', async () => {
      await input.trigger('keyup.esc')

      const onPressEscEvent = wrapper.emitted('onPressEsc')

      expect(wrapper.emitted()).toHaveProperty('onPressEsc')
      expect(onPressEscEvent).toHaveLength(1)
    })
  })
})
