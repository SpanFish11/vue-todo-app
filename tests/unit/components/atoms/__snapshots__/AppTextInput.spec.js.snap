// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AppTextInput component should render 1`] = `
<input
  type="text"
/>
`;

exports[`Given the AppTextInput component should render 2`] = `
<input
  type="text"
/>
`;
